frontend/src/views/Game.tsx

```diff
@@ -180,26 +192,35 @@ function GamePage() {
 
     // Get the result of promises and set the default code list.
     Promise.all(promises).then((result) => {
-      const codeList = [];
+      const newCodeList = [];
+      const newLanguageList = [];
+      const codeMap = result;
 
       for (let i = 0; i < result.length; i += 1) {
-        codeList.push(result[i][language[i]]);
+        newCodeList.push(result[i][Language.Python]);
+        newLanguageList.push(Language.Python);
       }
 
       // If previous code and language specified, save those as defaults
-      if (code) {
-        codeList[0] = code;
+      for (let i = 0; i < result.length; i += 1) {
+        const temp = getSubmission(i, playerSubmissions);
+
+        if (temp != null) {
+          newCodeList[i] = temp.code;
+          codeMap[i][temp.language as Language] = temp.code;
+          newLanguageList[i] = temp.language as Language;
+          setCurrentProblem(i);
+        }
       }
```

Comment:
So is this looping through all default code setups, and if there is a submission with that language and problem, store that in the local variables?

Reply:
This loops through the problems, and using getSubmission() it gets the latest submission made for that problem. From the submission, it grabs the code and language submitted, and sets the languageList, codeList, and codeMap accordingly. So what this really does is get the most recent submitted code for each problem from the submissions the backend passes back on reload, preserving code across reloads.

Reply:
Ah got it - thanks for the explanation! 👍